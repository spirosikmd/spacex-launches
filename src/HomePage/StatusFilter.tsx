import React, { SyntheticEvent } from 'react';
import PropTypes from 'prop-types';
import FormControl from '@material-ui/core/FormControl';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import FormGroup from '@material-ui/core/FormGroup';
import FormLabel from '@material-ui/core/FormLabel';
import Switch from '@material-ui/core/Switch';

interface StatusFilterProps {
  showFailed: boolean;
  showSuccessful: boolean;
  showUpcoming: boolean;
  onChange: (value: string, checked: boolean) => void;
}

const StatusFilter = ({
  showFailed,
  showSuccessful,
  showUpcoming,
  onChange,
}: StatusFilterProps) => {
  const handleChange = (event: SyntheticEvent) => {
    const target = event.currentTarget as HTMLInputElement;
    onChange(target.value, target.checked);
  };

  return (
    <FormControl component={'fieldset' as 'div'}>
      <FormLabel component={'legend' as 'label'}>Show launches</FormLabel>
      <FormGroup row>
        <FormControlLabel
          control={
            <Switch
              checked={showFailed}
              onChange={handleChange}
              value="showFailed"
            />
          }
          label="Failed"
        />
        <FormControlLabel
          control={
            <Switch
              checked={showSuccessful}
              onChange={handleChange}
              value="showSuccessful"
            />
          }
          label="Successful"
        />
        <FormControlLabel
          control={
            <Switch
              checked={showUpcoming}
              onChange={handleChange}
              value="showUpcoming"
            />
          }
          label="Upcoming"
        />
      </FormGroup>
    </FormControl>
  );
};

StatusFilter.propTypes = {
  showFailed: PropTypes.bool.isRequired,
  showSuccessful: PropTypes.bool.isRequired,
  showUpcoming: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default React.memo(StatusFilter);
