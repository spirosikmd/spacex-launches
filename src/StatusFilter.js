import React from 'react';
import PropTypes from 'prop-types';
import FormControl from '@material-ui/core/FormControl';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import FormGroup from '@material-ui/core/FormGroup';
import FormLabel from '@material-ui/core/FormLabel';
import Switch from '@material-ui/core/Switch';

export const StatusFilter = ({
  showFailed,
  showSuccessful,
  showUpcoming,
  onChange,
}) => {
  const handleChange = event => {
    onChange(event.currentTarget.value, event.currentTarget.checked);
  };

  return (
    <FormControl component="fieldset">
      <FormLabel component="legend">Show launches</FormLabel>
      <FormGroup row>
        <FormControlLabel
          control={
            <Switch
              checked={showFailed}
              onChange={handleChange}
              value="showFailed"
            />
          }
          label="Failed"
        />
        <FormControlLabel
          control={
            <Switch
              checked={showSuccessful}
              onChange={handleChange}
              value="showSuccessful"
            />
          }
          label="Successful"
        />
        <FormControlLabel
          control={
            <Switch
              checked={showUpcoming}
              onChange={handleChange}
              value="showUpcoming"
            />
          }
          label="Upcoming"
        />
      </FormGroup>
    </FormControl>
  );
};

StatusFilter.propTypes = {
  showFailed: PropTypes.bool.isRequired,
  showSuccessful: PropTypes.bool.isRequired,
  showUpcoming: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default React.memo(StatusFilter);
