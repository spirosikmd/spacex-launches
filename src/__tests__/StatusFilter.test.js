import FormControlLabel from '@material-ui/core/FormControlLabel';
import { StatusFilter } from '../StatusFilter';

describe('StatusFilter', () => {
  let props;

  beforeEach(() => {
    props = {
      showFailed: true,
      showSuccessful: true,
      showUpcoming: true,
      onChange: jest.fn(),
    };
  });

  it('renders status filters', () => {
    expect(shallow(StatusFilter, props)).toMatchSnapshot();
  });

  describe('when a status changes', () => {
    it('calls the onChange prop', () => {
      const wrapper = shallow(StatusFilter, props);
      const filters = wrapper.find(FormControlLabel);
      filters
        .at(0)
        .props()
        .control.props.onChange();
      expect(props.onChange).toHaveBeenCalled();
    });
  });
});
