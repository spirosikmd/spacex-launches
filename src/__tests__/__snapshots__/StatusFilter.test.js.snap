// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusFilter renders status filters 1`] = `
<WithStyles(FormControl)
  component="fieldset"
>
  <WithStyles(WithFormControlContext(FormLabel))
    component="legend"
  >
    Show launches
  </WithStyles(WithFormControlContext(FormLabel))>
  <WithStyles(FormGroup)
    row={true}
  >
    <WithStyles(WithFormControlContext(FormControlLabel))
      control={
        <WithStyles(Switch)
          checked={true}
          onChange={[MockFunction]}
          value="showFailed"
        />
      }
      label="Failed"
    />
    <WithStyles(WithFormControlContext(FormControlLabel))
      control={
        <WithStyles(Switch)
          checked={true}
          onChange={[MockFunction]}
          value="showSuccessful"
        />
      }
      label="Successful"
    />
    <WithStyles(WithFormControlContext(FormControlLabel))
      control={
        <WithStyles(Switch)
          checked={true}
          onChange={[MockFunction]}
          value="showUpcoming"
        />
      }
      label="Upcoming"
    />
  </WithStyles(FormGroup)>
</WithStyles(FormControl)>
`;
